account_invoice_product_supplier_price_update/account_invoice.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+# © 2016 Chafique DELLI @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api, _
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.model
+    def _check_product_supplierprice(self):
+        lines = []
+        for line in self.invoice_line:
+            current_price_unit = [(6, 0, [])]
+            suppinfo = self.env['product.supplierinfo'].search([
+                ('product_tmpl_id', '=', line.product_id.product_tmpl_id.id),
```

Comment:
you must depend on this module to get the right supplier https://github.com/OCA/product-variant/pull/32